tests/linearizability/failpoints.go

```diff
@@ -228,3 +230,23 @@ func (f randomFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Et
 func (f randomFailpoint) Name() string {
 	return "Random"
 }
+
+type memberAddFailpoint struct{}
+
+func (f memberAddFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+	// write more than SnapshotCount keys to single leader to make sure snapshot is created
+	testUserClientOpts := e2e.WithAuth("test-user", "abc")
+	for i := 0; i <= 10; i++ {
+		if err := clus.Client(testUserClientOpts).Put(ctx, "key", "test", config.PutOptions{}); err != nil {
+			panic(err)
+		}
+	}
+	if err := clus.StartNewProc(ctx, nil, t, e2e.WithAuth("root", "123")); err != nil {
```

Comment:
Not sure how running a new member is a trigger. Could we instead setup 3 member cluster, terminate one of the members (could be gradefully) and wait until there are enough transaction to trigger snapshot before starting it again?

Reply:
Yeah, I can try this out.